test/conftest.py

```diff
@@ -173,10 +180,22 @@ def fixture_ecr_image(docker_registry, docker_base_name, tag):
 
 
 @pytest.fixture(autouse=True)
-def skip_by_device_type(request, use_gpu, instance_type):
+def skip_by_device_type(request, use_gpu, instance_type, accelerator_type):
     is_gpu = use_gpu or instance_type[3] in ['g', 'p']
-    if (request.node.get_closest_marker('skip_gpu') and is_gpu) or \
-            (request.node.get_closest_marker('skip_cpu') and not is_gpu):
+    is_eia = accelerator_type is not None
+
+    # Separate out cases for clearer logic.
+    # When running GPU test, skip CPU test. When running CPU test, skip GPU test.
+    if (request.node.get_closest_marker('gpu_test') and not is_gpu) or \
+            (request.node.get_closest_marker('cpu_test') and is_gpu):
+        pytest.skip('Skipping because running on \'{}\' instance'.format(instance_type))
+
+    # When running EIA test, skip the CPU and GPU functions
+    elif (request.node.get_closest_marker('gpu_test') or request.node.get_closest_marker('cpu_test')) and is_eia:
```

Comment:
this is fine for now, but I don't think get_closest_marker will find the marker if someone decorates a pytest test with multiple fixutres like

```
@pytest.mark.fixture1
@pytest.mark.gpu_test
@pytest.mark.fixture2
def test_example():
```

Reply:
I agree, but `get_closest_marker()` is what's being used currently in master branch. I figured I would keep that there in case someone had a good reason for using it.
https://github.com/aws/sagemaker-pytorch-serving-container/blob/master/test/conftest.py#L176